desktop/plugins/network/RequestDetails.tsx

```diff
@@ -804,6 +805,85 @@ class BinaryFormatter {
   }
 }
 
+class ProtobufFormatter {
+  private protobufDefinitionRepository = ProtobufDefinitionsRepository.getInstance();
+
+  formatRequest(request: Request) {
+    if (
+      getHeaderValue(request.headers, 'content-type') ===
+      'application/x-protobuf'
+    ) {
+      const protobufDefinition = this.protobufDefinitionRepository.getRequestType(
+        request.method,
+        request.url,
+      );
+      if (protobufDefinition == undefined) {
+        return (
+          <Text>
+            Could not locate protobuf definition for request body of{' '}
+            {request.url}
+          </Text>
+        );
+      }
+
+      if (request?.data) {
+        const data = protobufDefinition.decode(
+          this._base64ToArrayBuffer(request.data),
+        );
+        return <JSONText>{data.toJSON()}</JSONText>;
+      } else {
+        return (
+          <Text>Could not locate request body data for {request.url}</Text>
+        );
+      }
+    }
+    return undefined;
+  }
+
+  formatResponse(request: Request, response: Response) {
+    if (
+      getHeaderValue(response.headers, 'content-type') ===
+        'application/x-protobuf' ||
+      request.url.endsWith('.proto')
+    ) {
+      const protobufDefinition = this.protobufDefinitionRepository.getResponseType(
+        request.method,
+        request.url,
+      );
+      if (protobufDefinition == undefined) {
+        return (
+          <Text>
+            Could not locate protobuf definition for response body of{' '}
+            {request.url}
+          </Text>
+        );
+      }
+
+      if (response?.data) {
+        const data = protobufDefinition.decode(
+          this._base64ToArrayBuffer(response.data),
+        );
+        return <JSONText>{data.toJSON()}</JSONText>;
+      } else {
+        return (
+          <Text>Could not locate response body data for {request.url}</Text>
+        );
+      }
+    }
+    return undefined;
+  }
+
+  _base64ToArrayBuffer(base64: string): Uint8Array {
+    const binary_string = window.atob(base64);
```

Comment:
`window.atob` is typically not what you can use if the base64 encodes binary data rather than an UTF-8 string, and it can crash Flipper pretty hard. Instead, you probably need [`Base64.toUint8Array`](https://www.npmjs.com/package/js-base64#synopsis) (we use the same package in the rest of the network plugin as well)